pkg/trtl/server.go

```diff
@@ -193,6 +204,14 @@ func (t *Server) Shutdown() (err error) {
 	log.Info().Msg("gracefully shutting down trtl server")
 	t.srv.GracefulStop()
 
+	// Shutdown the backup manager
+	if t.conf.Backup.Enabled {
+		if err = t.backup.Shutdown(); err != nil {
+			log.Error().Err(err).Msg("could not shutdown backup manager")
+			errs = append(errs, err)
+		}
+	}
```

Comment:
Do we have to add more concurrency control here? This is signaling the BackupManager to shutdown but I don't think it's waiting for it to finish up, which could cause problems (although unlikely) if the BackupManager is in the middle of doing a backup and the DB gets closed out from under it.